Userland/Libraries/LibCore/ArgsParser.cpp

```diff
@@ -484,6 +484,23 @@ void ArgsParser::add_option(Optional<double>& value, const char* help_string, co
     add_option(move(option));
 }
 
+void ArgsParser::add_option(Optional<size_t>& value, const char* help_string, const char* long_name, char short_name, const char* value_name)
+{
+    Option option {
+        true,
+        help_string,
+        long_name,
+        short_name,
+        value_name,
+        [&value](const char* s) {
+            auto opt = AK::StringUtils::convert_to_uint<size_t>(s);
+            value = opt.value_or(0);
+            return opt.has_value();
```

Comment:
We should match the behavior of the `Optional<bool>` overload above, otherwise `value` here will always have a value:
```c++
            value = AK::StringUtils::convert_to_uint<size_t>(s);
            return value.has_value();
```

Reply:
Fixed :+1: